sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_link_async.py

```diff
@@ -226,6 +232,9 @@ async def _outgoing_detach(self, close: bool = False, error: Optional[AMQPError]
         await self._session._outgoing_detach(detach_frame) # pylint: disable=protected-access
         if close:
             self._is_closed = True
+            self.links.pop(self.name, None)
```

Comment:
Had to add this here - see Test `test_queue_message_settle_through_mgmt_link_due_to_broken_receiver_link` was causing settlement over mgmt to not happen bc we weren't recognizing that the receiver link had been closed so the Transfer for settlement never gets sent and we infinitely loop while waiting on client_ready_async()

Reply:
yes and with closed=False you do get a corresponding incoming detach from the service